packages/passport-interface/README.md

```diff
@@ -1,3 +1,167 @@
-# `@pcd/passport-interface`
+<p align="center">
+    <h1 align="center">
+        @pcd/passport-interface
+    </h1>
+</p>
 
-This package contains code that defines the API of the passport server and client, and enables developers (both us, and third party developers) to communicate with those two applications. For example, this package contains code that enables you to construct a URL that requests a particular PCD from the passport webapp.
+<p align="center">
+    <a href="https://github.com/proofcarryingdata">
+        <img src="https://img.shields.io/badge/project-PCD-blue.svg?style=flat-square">
+    </a>
+    <a href="https://github.com/proofcarryingdata/zupass/blob/main/packages/passport-interface/LICENSE">
+        <img alt="License" src="https://img.shields.io/badge/license-GPL--3.0-green.svg?style=flat-square">
+    </a>
+    <a href="https://www.npmjs.com/package/@pcd/passport-interface">
+        <img alt="NPM version" src="https://img.shields.io/npm/v/@pcd/passport-interface?style=flat-square" />
+    </a>
+    <a href="https://npmjs.org/package/@pcd/passport-interface">
+        <img alt="Downloads" src="https://img.shields.io/npm/dm/@pcd/passport-interface.svg?style=flat-square" />
+    </a>
+    <a href="https://docs.pcd.team/modules/_pcd_passport_interface.html">
+        <img alt="Docs" src="https://img.shields.io/badge/docs-typedoc-purple.svg?style=flat-square">
+    </a>
+</p>
+
+| This package contains code that defines the API of the passport server and client, and enables developers (both us, and third party developers) to communicate with those two applications. For example, this package contains code that enables you to construct a URL that requests a particular PCD from the passport webapp. |
+| ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+
+
+## 🛠 Install
+
+Install the `@pcd/passport-interface` package with npm:
+
+```bash
+npm i @pcd/passport-interface
+```
+
+or yarn:
+
+```bash
+yarn add @pcd/passport-interface
+```
+
+## 📜 Usage
+
+The communication between users and passport webapp is mostly via a popup, which redirects to the webapp according to the URL and its search parameters in the same popup. Once the webapp generates the requested content, the popup automatically closes, and the resulting content can be accessed through a React hook.
+
+For more detailed information about the package's functions and types, please refer to the [documentation site](https://docs.pcd.team/modules/_pcd_passport_interface.html).
+
+### Using the passport popup
+
+You can use the `usePassportPopupSetup` React hook to set up necessary passport popup logic on a specific route (e.g. `/popup`) and communicate with the passport webapp.
+
```

Comment:
Please explain a little more verbosely why exactly this is needed. You mentioned also that you and Jei created a diagram that explains what's going on with the popups - may be worth adding here.